scipy/sparse/linalg/_eigen/lobpcg/lobpcg.py

```diff
@@ -67,20 +66,46 @@ def _makeMatMat(m):
         return lambda v: m @ v
 
 
+def _dot_inplace(x, y, verbosityLevel=0):
+    """Perform 'np.dot' in-place if dtypes match."""
```

Comment:
As I mentioned, I think this could be:
```suggestion
def _try_inplace(x, y, func, verbosityLevel=0):
    """
    Try inplace operation for dot or matmul.

    If some sufficient conditions for inplace operation are met, do so.
    Otherwise try inplace update and fall back to overwrite if that fails.
    `func` must be either np.dot or np.matmul.
    """
```

Reply:
OK, I see that everything just works with matmul. The docstring can still be improved as suggested (minus the part about `func`).

Reply:
Also the commented out parts of `np.dot` in the code need to be removed.

Reply:
Done